tests/integration/test_managers_integration.py

```
@@ -0,0 +1,439 @@
+        instance = manager.create_instance("component_50")
+        assert instance is not None
+    
+    def test_manager_memory_usage(self):
+        """Test manager memory usage."""
+        manager = DatasetManager()
+        
+        # Register and create many instances
+        for i in range(50):
+            class MockLoader:
+                def __init__(self):
+                    self.name = f"loader_{i}"
```

Ruff (0.13.1)

Function definition does not bind loop variable `i`

(B023)